src/main/scala/bi/deep/DruidSchemaReader.scala

```diff
@@ -37,26 +38,24 @@ case class DruidSchemaReader(config: Config) extends DruidSegmentReader {
         sparkSchema
       }
     }
-    else {
-      sparkSchema
-    }
   }
 }
 
 object DruidSchemaReader {
-  def readSparkSchema(qisa: QueryableIndexStorageAdapter): SparkSchema = SchemaUtils.druidToSpark(readDruidSchema(qisa))
+  def readSparkSchema(queryableIndexStorageAdapter: QueryableIndexStorageAdapter):
+  SparkSchema = SchemaUtils.druidToSpark(readDruidSchema(queryableIndexStorageAdapter))
 
-  private def readDruidSchema(qisa: QueryableIndexStorageAdapter): DruidSchema = {
+  private def readDruidSchema(queryableIndexStorageAdapter: QueryableIndexStorageAdapter): DruidSchema = {
 
-    val metrics = qisa.getAvailableMetrics.asScala.toArray
-    val dimensions = qisa.getAvailableDimensions.iterator().asScala.toArray
+    val metrics = queryableIndexStorageAdapter.getAvailableMetrics.asScala.toArray
+    val dimensions = queryableIndexStorageAdapter.getAvailableDimensions.iterator().asScala.toArray
 
     val dCapabilities = dimensions.map { dimension =>
-      (dimension, qisa.getColumnCapabilities(dimension).asInstanceOf[ColumnCapabilitiesImpl])
+      (dimension, queryableIndexStorageAdapter.getColumnCapabilities(dimension).asInstanceOf[ColumnCapabilitiesImpl])
     }
 
     val mCapabilities = metrics.map { metric =>
-      (metric, qisa.getColumnCapabilities(metric).asInstanceOf[ColumnCapabilitiesImpl])
+      (metric, queryableIndexStorageAdapter.getColumnCapabilities(metric).asInstanceOf[ColumnCapabilitiesImpl])
```

Comment:
You can also simply it as you can create array just at the end. Here we are dealing with iterators then let's leverage them and then call `toArray` also we can do it in one go. Also body is redundant as we to it twice, so:

```scala
val getCapabilities: String => (String, ColumnCapabilitiesImpl) = column => column -> druidStorage.getColumnCapabilities(name)

val metrics = druidStorage.getAvailableMetrics.asScala
    .map(getCapabilities).toArray

val dimensions = druidStorage.getAvailableDimensions.iterator().asScala
    .map(getCapabilities).toArray
```